README.md

```diff
@@ -228,8 +228,8 @@ If you'd like to contribute to the project or know the architecture and internal
    ```
 5. Because Codespaces gives a single externally routable host name for an instance, idpbuilder must deploy with path based routing. 
    This means ArgoCD and Gitea UIs are given with the following commands.
-   * ArgoCD: `echo https://${CODESPACE_NAME}-8080.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}/argocd`
-   * Gitea: `echo https://${CODESPACE_NAME}-8080.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}/gitea`
+   * ArgoCD: `echo https://argocd.${CODESPACE_NAME}-8080.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}`
+   * Gitea: `echo https://gitea.${CODESPACE_NAME}-8080.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}`
```

Comment:
Looks like it does need this change. 
I ran it is codespace which gave me this invalid link http://psychic-space-potato-gxq4q6pq4ppfwwxx-8080.app.github.dev:8080/argocd at the end to access argocd. 
But the same worked when I modified the link as http://argocd.psychic-space-potato-gxq4q6pq4ppfwwxx-8080.app.github.dev:8080
<img width="1361" alt="Screenshot 2024-05-29 at 9 49 47 AM" src="https://github.com/cnoe-io/idpbuilder/assets/21274175/2725a130-39f8-41fd-bf36-187e205782b2">

I added another commit to idpbuilder/pkg/cmd/create/root.go to fix this output url. Correct me if I'm missing anything here.

Reply:
Thanks for the detailed info. I just tried it in Codespaces through my browser (Firefox) and it does work as expected. The command used was:

```bash
./idpbuilder create --protocol http  \
    --host ${CODESPACE_NAME}-8080.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN} \
    --port 8080 --use-path-routing
```

I am surprised you can access argocd UI at `argocd.<GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN>`.  As far as I am aware, most browsers won't let you access it because of HSTS. Are you running this in your browser?

Reply:
You're right! `argocd.<GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN>` is not working due to HSTS in my browser. However, `<GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN>/argocd` is giving 404. How are you accessing these UIs?

Reply:
`https://${CODESPACE_NAME}-8080.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}/argocd` 

This should work. Note the port is the default HTTPS port 443 because this part is handled by Codespaces. My suggestion is to destroy everything and try it again.

```bash
# delete cluster. THIS WILL DELETE EVERYTHING IN CLUSTER
kind delete clusters localdev && docker system prune -f && docker volume prune -f
./idpbuilder create --protocol http --host ${CODESPACE_NAME}-8080.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}  --port 8080 --use-path-routing
```

Reply:
`https://${CODESPACE_NAME}-8080.${GITHUB_CODESPACES_PORT_FORWARDING_DOMAIN}/argocd
` is working for me as well.
Previously, I was using the URL provided at the end of the idpbuilder create command output (check screenshot [here](https://github.com/cnoe-io/idpbuilder/pull/267#discussion_r1619218997)), which had :8080 appended at the end. Looks like this is not needed with path based routing in Codespace. I have updated the PR to reflect this finding.

Reply:
The changes to code has some unintended consequences. For example, if you are running it on your local machine with path based routing, the printed URL is incorrect:

Expected: `https://cnoe.localtest.me:8443/argocd`
Reality:  `https://cnoe.localtest.me/argocd`

If you are running this in environment where it's fronted by another proxy like Codespaces, what you did is correct. But we can't assume we are fronted by another proxy whenever the `--use-path-routing` flag is used. This flag is valid for local machine use cases as well.

I agree that the URL printed out at the end is incorrect for Codespaces and should be fixed. Proper fix for this will involve detecting if we have a proxy in front of us. 

I think we can merge this PR with changes to the README file only. We can tackle the codespaces problem in another PR and issue.

Reply:
Agreed! I din't think of that scenario. removed those changes from this PR